utils/helpers.ts

```diff
@@ -151,3 +151,62 @@ export function checkSearchTerm(searchTerm: string, plant: Plant) {
   // Check if plant_name contains searchTerm
   return plant.plant_name.toLowerCase().includes(searchTerm.toLowerCase());
 }
+
+export function checkSunlight(
+  sunlightFilterValue: DropdownOption[],
+  plant: Plant,
+) {
+  // Automatically returns true if no selected sunlight
+  if (sunlightFilterValue.length === 0) {
+    return true;
+  }
+
+  const sunlightToHours = new Map<string, [number, number]>([
+    ['SHADE', [0, 2]],
+    ['PARTIAL_SHADE', [2, 4]],
+    ['PARTIAL_SUN', [4, 6]],
+    ['FULL', [6, Number.MAX_VALUE]],
+  ]);
+
+  // For each sunlight selected, check if plant's min_hours and max_hours are
+  // within that enum's range
+  const sunlightBoolean: boolean[] = [];
+  for (const sunlight of sunlightFilterValue) {
+    const [minHours, maxHours] = sunlightToHours.get(sunlight.value)!;
+    // if max_hours is null then plant can only receive min_hours of sunlight
+    if (plant.sunlight_max_hours === null) {
+      sunlightBoolean.push(
+        plant.sunlight_min_hours >= minHours &&
+          plant.sunlight_min_hours <= maxHours,
+      );
+    } else {
+      sunlightBoolean.push(
+        plant.sunlight_min_hours >= minHours &&
+          plant.sunlight_max_hours <= maxHours,
+      );
+    }
+  }
+
+  // Return true if any of the sunlightBooleans are true
+  return sunlightBoolean.includes(true);
+}
+
+export function checkDifficulty(
+  difficultyFilterValue: DropdownOption[],
+  plant: Plant,
+) {
+  // Automatically returns true if no selected difficulty
+  if (difficultyFilterValue.length === 0) {
+    return true;
+  }
+
+  // For each difficulty selected,
+  // check if plant's difficulty_level matches difficulty
+  const difficultyBoolean: boolean[] = [];
```

Comment:
same comment about the arrays - just noting it here
Honestly, i think it's ok to leave it as it is now, since the logic is correct, but I just wanted to flag this in case we wanted to change it in the future.